specification/cost-management/resource-manager/Microsoft.CostManagement/stable/2022-10-01/costmanagement.benefits.json

```diff
@@ -0,0 +1,1002 @@
+        }
+      }
+    },
+    "/providers/Microsoft.BillingBenefits/savingsPlanOrders/{savingsPlanOrderId}/providers/Microsoft.CostManagement/benefitUtilizationSummaries": {
+      "get": {
+        "tags": [
+          "BenefitUtilizationSummaries"
+        ],
+        "operationId": "BenefitUtilizationSummaries_ListBySavingsPlanOrder",
+        "description": "Lists the savings plan utilization summaries for daily or monthly grain.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/cost-management/"
+        },
+        "x-ms-examples": {
+          "SavingsPlanUtilizationSummariesDaily": {
+            "$ref": "./examples/BenefitUtilizationSummaries/SavingsPlan-SavingsPlanOrderId-Daily.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "$filter",
+            "description": "Supports filtering by usageDate.",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/grainParameter"
+          },
+          {
+            "$ref": "#/parameters/savingsPlanOrderIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/BenefitUtilizationSummariesListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common-types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.BillingBenefits/savingsPlanOrders/{savingsPlanOrderId}/savingsPlans/{savingsPlanId}/providers/Microsoft.CostManagement/benefitUtilizationSummaries": {
+      "get": {
+        "tags": [
+          "BenefitUtilizationSummaries"
+        ],
+        "operationId": "BenefitUtilizationSummaries_ListBySavingsPlanId",
+        "description": "Lists the savings plan utilization summaries for daily or monthly grain.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/cost-management/"
+        },
+        "x-ms-examples": {
+          "SavingsPlanUtilizationSummariesMonthlyWithSavingsPlanId": {
+            "$ref": "./examples/BenefitUtilizationSummaries/SavingsPlan-SavingsPlanId-Monthly.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "$filter",
+            "description": "Supports filtering by usageDate.",
```

Comment:
>   "description": "Supports filtering by usageDate.", [](http://example.com/codeflow?start=9&length=53)

Should it be modeled as an enum? Since its a finite list of values you're allowing.

Reply:
Not addressing this suggestion, since I want to keep the filter definition consistent with Cost management RP. Also, in the code we plan to support using ODataQueryOptions. So in the future, we may end up supporting filters on all the properties.